src/Resources/public/js/search.js

```diff
@@ -43,43 +43,115 @@ class SearchManager {
+        if (typeof form === 'string') {
             form = document.querySelector(form);
 
-            if(null === form) {
+            if (null === form) {
                 throw new Error('Form not found');
             }
         }
 
         this.#form = form;
 
-        this.#form.addEventListener('input', function (event) {
-            /** @type {HTMLInputElement} */
+        this.#form.addEventListener('input', (event) => {
             const field = event.target;
 
+            if (!(field instanceof HTMLInputElement || field instanceof HTMLSelectElement)) {
+                return;
+            }
+
             field.dispatchEvent(new CustomEvent('search:form-changed', { bubbles: true }));
 
             // todo would be better to have a way to check the type of the field directly on the field, e.g. with a data attribute
-            if(field.closest('.ssm-filter') !== null) {
+            if (field.closest('.ssm-filter') !== null) {
                 field.dispatchEvent(new CustomEvent('search:filter-changed', { bubbles: true }));
-            } else if(field.name === 'p') {
+            } else if (field.name === 'p') {
                 field.dispatchEvent(new CustomEvent('search:page-changed', { bubbles: true }));
-            } else if(field.classList.contains('ssm-sort')) {
+            } else if (field.classList.contains('ssm-sort')) {
                 field.dispatchEvent(new CustomEvent('search:sort-changed', { bubbles: true }));
             }
         });
 
-        this.#form.addEventListener('search:filter-changed', (event) => this.#options.onFilterChange.bind(this, this.#form, event.target)());
-        this.#form.addEventListener('search:page-changed', (event) => this.#options.onSortChange.bind(this, this.#form, event.target)());
-        this.#form.addEventListener('search:sort-changed', (event) => this.#options.onSortChange.bind(this, this.#form, event.target)());
-        this.#form.addEventListener('submit', () => this.#options.onSubmit.bind(this, this.#form)());
+        this.#form.addEventListener('search:filter-changed', (event) => this.#options.onFilterChange(this.#form, event.target));
+        this.#form.addEventListener('search:page-changed', (event) => this.#options.onPageChange(this.#form, event.target));
+        this.#form.addEventListener('search:sort-changed', (event) => this.#options.onSortChange(this.#form, event.target));
+        this.#form.addEventListener('submit', (event) => {
+            event.preventDefault();
+            this.#updateURL();
+            this.#options.onSubmit(this.#form);
+        });
+    }
+
+    /**
+     * @param {HTMLFormElement} form
+     */
+    async #submitForm(form) {
+        this.#options.loader.show(this.#options.loader.selector);
+        this.disableEmptyFields();
+
+        try {
+            const requestOptions = {
+                method: form.method.toUpperCase(),
+            };
+
+            if (requestOptions.method === 'POST') {
```

Comment:
Will it ever be `POST`?

Reply:
The URL max length may be reached.

Reply:
Okay, but we don't handle that anyway, so I guess for now this is not needed?

Reply:
If someone will handle that in their project then they'll have to change the JS too.